src/main/java/com/jpaplayground/domain/product/Product.java

```diff
@@ -1,34 +1,55 @@
 package com.jpaplayground.domain.product;
 
+import java.time.LocalDateTime;
+import javax.persistence.Column;
 import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
 import javax.persistence.GeneratedValue;
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
 import lombok.AccessLevel;
 import lombok.Builder;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
 
 @Entity
 @Getter
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
+@EntityListeners(AuditingEntityListener.class)
 public class Product {
 
 	@Id
 	@GeneratedValue(strategy = GenerationType.IDENTITY)
 	private Long id;
+
 	private String name;
+
 	private Integer price;
+
 	private Boolean deleted;
 
+	@Column(updatable = false)
+	@CreatedDate
+	private LocalDateTime createdAt;
+
+	@LastModifiedDate
+	private LocalDateTime lastModifiedAt;
+
+	@Column(updatable = false)
+	private Long creatorId;
```

Comment:
이전에 AuditorAware를 구현하였을 때 1. **request scoped Bean**(`LoginMember`)을 사용하는 방법과 2. **HttpServletRequest** + request attribute을 이용하는 방법을 사용해봤었습니다. 그런데 두 방법 모두 제가 정확히 이해를 못한 점이 있어서 삭제했었는데, 마침 얘기가 나왔으니 질문드립니닷

```
@Component
@RequiredArgsConstructor
public class MemberAuditorAware implements AuditorAware<Member> {

	private final HttpServletRequest request;

	@Override
	public Optional<Member> getCurrentAuditor() {
		~~~
	}
}
```

1. HttpServletRequest도 위와 같이 주입이 되더군요! 이 경우 request scoped Bean과 같은 역할을 하는 것으로 보면 될까요?  

2. singleton으로 관리되는 Bean들과 달리 request scoped Bean은 request 마다 새로 생성되는 것으로 알고있습니다.
위와 같은 클래스는 singleton으로 하나만 생성되고 모든 request가 공유하게 될텐데, 공유변수를 둔 것 마냥 충돌이 발생하지 않나요?
(이 고민 때문에 Auditor를 삭제하였었습니다)

3. 그게 아니라면 어떻게 충돌이 나지 않고 처리되는건지 궁금합니다. 

음.. 결국 Spring의 싱글톤이 어떻게 여러 요청을 처리하는지를 제대로 이해하지 못하고 있어서 생긴 질문들이네요.

'같은 클래스를 Block하면서 사용하는게 아니라, 메소드 영역에 저장된 그 클래스의 로직만 사용하는거다' 정도로만 일단 이해하였는데, 어렵네요..
📚 참고:
https://jeong-pro.tistory.com/204
https://okky.kr/articles/163660

Reply:
서블릿 컨테이너는 HTTP 요청 마다 별도의 쓰레드를 제공하기 때문에 같은 HTTP 요청이면 같은 스프링 빈이 반환되도록 되어있습니다.

Reply:
>AuditorAware은 Spring 싱글톤 Bean 동작 원리에 대해 조금 더 이해한 뒤에 적용해보도록 하겠습니다

이전 코멘트와 더불어 스프링 애플리케이션을 실행 시 request 스코프 빈 주입과 관련된 아래의 에러 메시지와

`Scope 'request' is not active for the current thread`

아래 클래스에 대한 설명을 읽어보시면 감이 잡히실 거 같습니다

```java
public interface BoundContext<S>
 extends WeldAlterableContext
```
Allows a thread-based context to be bound to some external instance storage (such as an HttpSession).

Reply:
>AuditorAware은 Spring 싱글톤 Bean 동작 원리에 대해 조금 더 이해한 뒤에 적용해보도록 하겠습니다

이전 코멘트와 더불어 스프링 애플리케이션을 실행 시 request 스코프 빈 주입과 관련된 아래의 에러 메시지와

`Scope 'request' is not active for the current thread`

아래 클래스에 대한 설명을 읽어보시면 감이 잡히실 거 같습니다

```java
public abstract class RequestContextHolder
 extends java.lang.Object
```
Holder class to expose the web request in the form of a thread-bound RequestAttributes object.

Reply:
>AuditorAware은 Spring 싱글톤 Bean 동작 원리에 대해 조금 더 이해한 뒤에 적용해보도록 하겠습니다

이전 코멘트와 더불어 스프링 애플리케이션을 실행 시 request 스코프 빈 주입과 관련된 아래의 에러 메시지와

`Scope 'request' is not active for the current thread`

아래 클래스에 대한 설명을 읽어보시면 감이 잡히실 거 같습니다

```java
public interface RequestContext
```

The web flow system will ensure that a RequestContext object is local to the current thread. It can be safely manipulated without needing to worry about concurrent access.

Reply:
>AuditorAware은 Spring 싱글톤 Bean 동작 원리에 대해 조금 더 이해한 뒤에 적용해보도록 하겠습니다

이전 코멘트와 더불어 스프링 애플리케이션을 실행 시 request 스코프 빈 주입과 관련된 아래의 에러 메시지와

`Scope 'request' is not active for the current thread`

아래 인터페이스에 대한 설명을 읽어보시면 감이 잡히실 거 같습니다

```java
public interface RequestContext
```

The web flow system will ensure that a RequestContext object is local to the current thread. It can be safely manipulated without needing to worry about concurrent access.

Reply:
새 요청이 들어올 때마다 별도의 쓰레드가 생성되고, `request scoped bean`이나 `HttpServletRequest`는 그 쓰레드 마다 고유한 인스턴스가 생성되므로 `Autowired`로 주입받아 사용하여도 동시성 이슈는 걱정하지 않아도 되겠군요!
설명 감사드립니다!! 👍 👍